File: typescript/tests/integration/network.integration.test.ts
import { AgentBay, CreateSessionParams } from "../../src";

describe("BetaNetwork integration", () => {
  test("create/describe beta network and bind session via betaNetworkId", async () => {
    const apiKey = process.env.AGENTBAY_API_KEY;
    if (!apiKey) {
      return;
    }

    const client = new AgentBay({ apiKey });

    const netResult = await client.betaNetwork.getNetworkBindToken();
    expect(netResult.success).toBe(true);
    expect(netResult.networkId).not.toBe("");
    expect(netResult.networkToken).not.toBe("");

    const statusResult = await client.betaNetwork.describe(netResult.networkId);
    expect(statusResult.success).toBe(true);

    const params: CreateSessionParams = {
      imageId: "imgc-0ab5takhjgjky7htu",
      betaNetworkId: netResult.networkId,
      labels: { "test-type": "network-integration" }
    };

    const createResult = await client.create(params);
    expect(createResult.success).toBe(true);
    expect(createResult.session).toBeDefined();
    if (!createResult.session) {
      return;
    }

    await createResult.session.delete();
  });
});

File: golang/tests/pkg/integration/network_integration_test.go
package integration_test

import (
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
	"github.com/aliyun/wuying-agentbay-sdk/golang/tests/pkg/agentbay/testutil"
)

func TestNetworkCreateDescribeAndBindSession(t *testing.T) {
	apiKey := testutil.GetTestAPIKey(t)

	client, err := agentbay.NewAgentBay(apiKey)
	if err != nil {
		t.Fatalf("Error initializing AgentBay client: %v", err)
	}

	if client.BetaNetwork == nil {
		t.Fatalf("client.BetaNetwork is nil")
	}

	networkResult, err := client.BetaNetwork.BetaGetNetworkBindToken("")
	if err != nil {
		t.Fatalf("BetaGetNetworkBindToken failed: %v", err)
	}
	if !networkResult.Success {
		t.Fatalf("BetaGetNetworkBindToken returned success=false: %s", networkResult.ErrorMessage)
	}
	if networkResult.NetworkId == "" {
		t.Fatalf("expected non-empty networkId")
	}
	if networkResult.NetworkToken == "" {
		t.Fatalf("expected non-empty networkToken")
	}

	statusResult, err := client.BetaNetwork.BetaDescribe(networkResult.NetworkId)
	if err != nil {
		t.Fatalf("BetaDescribe failed: %v", err)
	}
	if !statusResult.Success {
		t.Fatalf("BetaDescribe returned success=false: %s", statusResult.ErrorMessage)
	}

	params := agentbay.NewCreateSessionParams().
		WithImageId("linux_latest").
		WithBetaNetworkId(networkResult.NetworkId).
		WithLabels(map[string]string{"test-type": "network-integration"})

	createResult, err := client.Create(params)
	if err != nil {
		t.Fatalf("Create session failed: %v", err)
	}
	if createResult.Session == nil {
		t.Fatalf("createResult.Session is nil")
	}
	defer func() {
		_, _ = client.Delete(createResult.Session)
	}()
}
